.repo/sample-list-generator/src/libs/api-detector.ts

```diff
@@ -0,0 +1,174 @@
+import { ApiItem, ApiTypeResult, ExtensionApiMap } from '../types';
+import * as babel from '@babel/core';
+import fs from 'fs/promises';
+import { getAllFiles } from '../utils/filesystem';
+import { singularize } from '../utils/string';
+import { loadExtensionApis } from './api-loader';
+import os from 'os';
+import { ParallelController } from '../utils/parallel';
+
+const CPU_CORES = os.cpus().length;
+
+let EXTENSION_API_MAP: ExtensionApiMap = loadExtensionApis();
+
+export const getApiType = (
+  apiCategory: string,
+  apiName: string
+): ApiTypeResult => {
+  apiCategory = apiCategory.replace(/_/g, '.');
+
+  if (EXTENSION_API_MAP[apiCategory]) {
+    const apiTypes = EXTENSION_API_MAP[apiCategory];
+
+    for (let type of Object.keys(apiTypes)) {
+      if (apiTypes[type].includes(apiName)) {
+        return type as ApiTypeResult;
+      }
+    }
+  }
+  console.log('api not found', apiCategory, apiName);
+  return 'unknown';
+};
+
+const getNextLevelProperty = (
```

Comment:
Hmm, I'm not sure what this does. Could you make that clearer in the code with some comments and maybe renaming the function?